src/server/cluster/cluster_config.h

```diff
@@ -81,6 +81,9 @@ class ClusterConfig {
   static std::shared_ptr<ClusterConfig> CreateFromConfig(std::string_view my_id,
                                                          const JsonType& json_config);
 
+  std::shared_ptr<ClusterConfig> CloneAndUpdate(const std::vector<SlotRange>& slots,
```

Comment:
please add comment what this function do

Reply:
the naming AndUpdate is a bit confusing here as it soulds like you are updating the existing conifg. I would call this CreateFromExistingWithChanges or similar

Reply:
CloneWithChanges?